src/metabase/api/alert.clj

```diff
@@ -0,0 +1,182 @@
+(ns metabase.api.alert
+  "/api/alert endpoints"
+  (:require [clojure.data :as data]
+            [compojure.core :refer [DELETE GET POST PUT]]
+            [medley.core :as m]
+            [metabase
+             [email :as email]
+             [util :as u]]
+            [metabase.api
+             [common :as api]
+             [pulse :as pulse-api]]
+            [metabase.email.messages :as messages]
+            [metabase.models
+             [interface :as mi]
+             [pulse :as pulse :refer [Pulse]]]
+            [metabase.util.schema :as su]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(api/defendpoint GET "/"
+  "Fetch all alerts"
+  []
+  (for [alert (pulse/retrieve-alerts)
+        :let  [can-read?  (mi/can-read? alert)
+               can-write? (mi/can-write? alert)]
+        :when (or can-read?
+                  can-write?)]
+    (assoc alert :read_only (not can-write?))))
+
+(api/defendpoint GET "/question/:id"
+  [id]
```

Comment:
It's fairly clear what this does but I think the defendpoint macro is still set up to log a warning message if you don't add a docstring to an endpoint. It doesn't trigger the linter because then macro also appends it's own documentation anyway 

Btw when we're adding a bunch of API endpoints it's not a bad idea to run `lein run api-documentation` to update the generated dox. We should probably get that set up to run automatically at some point because I'm always forgetting

Reply:
Didn't know about `lein run api-documentation`, just ran it and it had a bunch of changes, I'll include it in this PR 👍